contracts/LockedFund.sol

```diff
@@ -0,0 +1,510 @@
+		isAdmin[_newAdmin] = true;
+
+		emit AdminAdded(msg.sender, _newAdmin);
+	}
+
+	/**
+	 * @notice Internal function to remove an admin.
+	 * @param _adminToRemove The address of the admin which should be removed.
+	 * @dev Only callable by an Admin.
+	 */
+	function _removeAdmin(address _adminToRemove) internal {
+		require(isAdmin[_adminToRemove], "LockedFund: Address is not an admin.");
+		isAdmin[_adminToRemove] = false;
+
+		emit AdminRemoved(msg.sender, _adminToRemove);
+	}
+
+	/**
+	 * @notice Internal function to update the Vesting Registry, Duration and Cliff.
+	 * @param _vestingRegistry The Vesting Registry Address.
+	 */
+	function _changeVestingRegistry(address _vestingRegistry) internal {
+		require(_vestingRegistry != address(0), "LockedFund: Vesting registry address is invalid.");
+
+		vestingRegistry = IVestingRegistry(_vestingRegistry);
+
+		emit VestingRegistryUpdated(msg.sender, _vestingRegistry);
+	}
+
+	/**
+	 * @notice Internal function used to update the waitedTS.
+	 * @param _waitedTS The timestamp after which withdrawal is allowed.
+	 */
+	function _changeWaitedTS(uint256 _waitedTS) internal {
+		require(_waitedTS != 0, "LockedFund: Waited TS cannot be zero.");
+
+		waitedTS = _waitedTS;
+
+		emit WaitedTSUpdated(msg.sender, _waitedTS);
+	}
+
+	/**
+	 * @notice Internal function to add Token to the user balance (Vested and Waited Unlocked Balance based on `_basisPoint`).
+	 * @param _userAddress The user whose locked balance has to be updated with `_amount`.
+	 * @param _amount The amount of Token to be added to the locked and/or unlocked balance.
+	 * @param _cliff The cliff for vesting.
+	 * @param _duration The duration for vesting.
+	 * @param _basisPoint The % (in Basis Point)which determines how much will be unlocked immediately.
+	 */
+	function _depositVested(
+		address _userAddress,
+		uint256 _amount,
+		uint256 _cliff,
+		uint256 _duration,
+		uint256 _basisPoint
+	) internal {
+		/// If duration is also zero, then it is similar to Unlocked Token.
+		require(_duration != 0, "LockedFund: Duration cannot be zero.");
+		require(_duration < MAX_DURATION, "LockedFund: Duration is too long.");
+
+		// MAX_BASIS_POINT is not included because if 100% is unlocked, then this function is not required to be used.
+		require(_basisPoint < MAX_BASIS_POINT, "LockedFund: Basis Point has to be less than 10000.");
+		bool txStatus = token.transferFrom(msg.sender, address(this), _amount);
+		require(txStatus, "LockedFund: Token transfer was not successful. Check receiver address.");
+
+		uint256 waitedUnlockedBal = _amount.mul(_basisPoint).div(MAX_BASIS_POINT);
+
+		waitedUnlockedBalances[_userAddress] = waitedUnlockedBalances[_userAddress].add(waitedUnlockedBal);
+		vestedBalances[_userAddress] = vestedBalances[_userAddress].add(_amount).sub(waitedUnlockedBal);
+
+		cliff[_userAddress] = _cliff * INTERVAL;
+		duration[_userAddress] = _duration * INTERVAL;
```

Comment:
new deposit overrides previous deposit settings.
is it the reason the function has `onlyOwner` access?

Reply:
The current vesting registry (New registry is currently being tested, might update this with that in the future) which is used can only accommodate one type of cliff and duration combination for a particular address, thus it is required to be limited to be called only by an owner, thus limiting any attack on this by creating a big cliff/duration, etc.